### PR TITLE
[ODS-4289] Convert GenerateSecurityGraphs to NET Core

### PR DESCRIPTION
**Pull request by raghuramakrishna**

### CODE REVIEW

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="CommandLineParser" Version="2.8.0" />
+    <PackageReference Include="QuikGraph" Version="2.2.0" />
+    <PackageReference Include="QuikGraph.Graphviz" Version="2.2.0" />
+    <PackageReference Include="QuikGraph.Serialization" Version="2.2.2" />
```
Comment (jamessmckay) [inline Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj:12]: Why are you not using the Sandwych.QuickGraph.Core package?

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    <PackageReference Include="System.Data.SqlClient" Version="4.8.2" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="Assets\AssessmentMetadata_icon.png">
```
Comment (jamessmckay) [inline Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj:17]: why are we adding these images? are they not generated?

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/Models/AuthorizationMetadata/EffectiveActionAndStrategy.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+        public string AuthorizationStrategy { get; private set; }
+
+        public string AuthorizationStrategyOriginatingResourceName { get; private set; }
+
+        public bool AuthorizationStrategyInherited { get; private set; }
+
+        public bool AuthorizationStrategyIsDefault { get; private set; }
+
+        //public bool AuthorizationStrategyDefaulted { private set; get; }
```
Comment (jamessmckay) [inline Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/Models/AuthorizationMetadata/EffectiveActionAndStrategy.cs:32]: this should be removed if not used

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/Models/Query/ClaimsetResourceActionData.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        //            //return "NoFurtherAuthorizationRequired";
+        //        }
+
+        //        return null;
+        //    }
+        //    set
+        //    {
+        //        // Do nothing
+        //    }
+        //}
```
Comment (jamessmckay) [inline Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/Models/Query/ClaimsetResourceActionData.cs:37]: if this is not used then remove it please

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/Program.cs [hunk 5]
```diff
@@ -0,0 +1,963 @@
...
+            'u', "user", HelpText =
+                "The SQL Server username to use for connecting to the authorization metadata database.  Leave username and password blank to use integrated security.")]
+        public string User { get; set; }
+
+        [Option(
+            'p', "password", HelpText =
+                "The password to use for connecting to the authorization metadata database.  Leave username and password blank to use integrated security.")]
+        public string Password { get; set; }
+
+        //[HelpOption]
```
Comment (jamessmckay) [inline Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/Program.cs:61]: remove this please

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/Program.cs [hunk 6]
```diff
@@ -0,0 +1,963 @@
...
+                            new AuthorizationKey(claimSetName, resource.Name, actionName),
+                            GetEffectiveActionPermissions(resource, actionName, claimSetName)
+                        );
+                    }
+                }
+            }
+
+            var rootNodes = GetRootNodes(resourceGraph);
+
+            //ClearOutputFolder();
```
Comment (jamessmckay) [inline Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/Program.cs:175]: should this be commented?

**Review by jamessmckay — CHANGES_REQUESTED** [review]
Why are all these images added?, please remove them and add a gitignore file that excludes the output file.  Also please move the dead code if it is not used, and/or restore it. Finally, the package you have chosen for quikgraph seems to be a deviation on the already approved package Sandwych.QuickGraph.Core. Please explain why this package is chosen.

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj (hunk 1 above)
Comment (raghuramakrishna) [reply Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj:12]: The package doesn't support DotEngine, so I found this package which works on minimal code changes.

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    <PackageReference Include="QuikGraph" Version="2.2.0" />
+    <PackageReference Include="QuikGraph.Graphviz" Version="2.2.0" />
+    <PackageReference Include="QuikGraph.Serialization" Version="2.2.2" />
+    <PackageReference Include="System.Data.SqlClient" Version="4.8.2" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="Assets\" />
+  </ItemGroup>
+
```
Comment (jamessmckay) [inline Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj:19]:
```suggestion
```

File: Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj [hunk 8]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (jamessmckay) [inline Utilities/GenerateSecurityGraphs/GenerateSecurityGraphs.NetCore/GenerateSecurityGraphs.NetCore.csproj:5]:
```suggestion
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <Copyright>Copyright © 2020 Ed-Fi Alliance, LLC and Contributors</Copyright>
    <TreatErrorsAsWarning>true</TreatErrorsAsWarning>
    <RestorePackages>true</RestorePackages>
```

**Review by jamessmckay — CHANGES_REQUESTED** [review]

**Review by jamessmckay — APPROVED** [review]

**Pull request merged**